File: fundamental/LatihanCustomView/java/app/src/main/java/com/dicoding/picodiploma/mycustomview/MyButton.java
package com.dicoding.picodiploma.mycustomview;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.drawable.Drawable;

import androidx.core.content.ContextCompat;

import android.util.AttributeSet;
import android.view.Gravity;

public class MyButton extends androidx.appcompat.widget.AppCompatButton {

    private Drawable enabledBackground, disabledBackground;

    private int textColor;

    // Konstruktor dari MyButton
    public MyButton(Context context) {
        super(context);
        init();
    }

    public MyButton(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public MyButton(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    // Metode onDraw() digunakan untuk mengcustom button ketika enable dan disable
    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);
        // Mengubah background dari Button
        setBackground(isEnabled() ? enabledBackground : disabledBackground);

        // Mengubah warna text pada button
        setTextColor(textColor);

        // Mengubah ukuran text pada button
        setTextSize(12.f);

        // Menjadikan object pada button menjadi center
        setGravity(Gravity.CENTER);

        // Mengubah text pada button pada kondisi enable dan disable
        setText(isEnabled() ? "Submit" : "Isi Dulu");
    }

    // pemanggilan Resource harus dilakukan saat kelas MyButton diinisialisasi, jadi harus dikeluarkan dari metode onDraw
    private void init() {
        textColor = ContextCompat.getColor(getContext(), android.R.color.background_light);
        enabledBackground = ContextCompat.getDrawable(getContext(), R.drawable.bg_button);
        disabledBackground = ContextCompat.getDrawable(getContext(), R.drawable.bg_button_disable);
    }
}

File: layout/LatihanCustomView/kotlin/app/src/main/java/com/dicoding/picodiploma/mycustomview/MyButton.kt
package com.dicoding.picodiploma.mycustomview

import android.content.Context
import android.graphics.Canvas
import android.graphics.drawable.Drawable
import android.util.AttributeSet
import android.view.Gravity
import androidx.appcompat.widget.AppCompatButton
import androidx.core.content.ContextCompat

class MyButton : AppCompatButton {

    private var enabledBackground: Drawable? = null
    private var disabledBackground: Drawable? = null

    private var txtColor: Int = 0

//  Konstruktor dari MyButton
    constructor(context: Context) : super(context) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        init()
    }

    // Metode onDraw() digunakan untuk mengcustom button ketika enable dan disable
    override fun onDraw(canvas: Canvas) {
        super.onDraw(canvas)
        // Mengubah background dari Button
        background = when {
            isEnabled -> enabledBackground
            else -> disabledBackground
        }

        // Mengubah warna text pada button
        setTextColor(txtColor)

        // Mengubah ukuran text pada button
        textSize = 12f

        // Menjadikan object pada button menjadi center
        gravity = Gravity.CENTER

        // Mengubah text pada button pada kondisi enable dan disable
        text = when {
            isEnabled -> "Submit"
            else -> "Isi Dulu"
        }
    }

    // pemanggilan Resource harus dilakukan saat kelas MyButton diinisialisasi, jadi harus dikeluarkan dari metode onDraw
    private fun init() {
        txtColor = ContextCompat.getColor(context, android.R.color.background_light)
        enabledBackground = ContextCompat.getDrawable(context, R.drawable.bg_button)
        disabledBackground = ContextCompat.getDrawable(context, R.drawable.bg_button_disable)
    }
}
